src/core/utils/variant.h

```diff
@@ -0,0 +1,24 @@
+#ifndef UTIL_VARIANT_H
+#define UTIL_VARIANT_H
+
+#include <variant>
+
+template <typename T, typename U>
```

Comment:
Nice solution for the specific use case of handling alternative formatting types (JSON vs. CSV/TSV). Seems like templated variant classes could make it difficult to extend for more types (e.g. if XML support was also added).

Reply:
What are you thinking?

I was thinking that we could extend this by create (for example) a struct XMLFormat, and adding it to the variant.  Then we'd have:
```
typedef std::variant<SeparatorFormat,JSONFormat,XMLFormat> SampleFormat;
```
Code that does `to<SeparateFormat>(format)` should be unaffected.